torch/testing/_internal/optests/generate_tests.py

```diff
@@ -0,0 +1,319 @@
+import functools
+import unittest
+
+import torch
+
+import torch.utils._pytree as pytree
+
+from torch._dynamo.utils import clone_input
+from torch._subclasses.schema_check_mode import SchemaCheckMode
+from torch.overrides import TorchFunctionMode
+from torch.testing._internal.optests import (
+    aot_autograd_check,
+    autograd_registration_check,
+    fake_check,
+)
+
+
+def safe_schema_check(op, args, kwargs):
+    args, kwargs = pytree.tree_map(clone_input, (args, kwargs))
+    with SchemaCheckMode():
+        result = op(*args, **kwargs)
+        return result
+
+
+def safe_autograd_registration_check(op, args, kwargs):
+    # Don't perform autograd_registration_check if none of the inputs require grad.
+    if not pytree.tree_any_only(
+        torch.Tensor, lambda x: x.requires_grad, (args, kwargs)
+    ):
+        return
+    args, kwargs = pytree.tree_map(clone_input, (args, kwargs))
+    return autograd_registration_check(op, args, kwargs)
+
+
+def safe_fake_check(op, args, kwargs):
+    args, kwargs = pytree.tree_map(clone_input, (args, kwargs))
+    return fake_check(op, args, kwargs, dynamic_only=False)
+
+
+def safe_aot_autograd_check(op, args, kwargs, dynamic):
+    def func(*args, **kwargs):
+        args, kwargs = pytree.tree_map_only(torch.Tensor, torch.clone, (args, kwargs))
```

Comment:
If there are any tests internally that rely on sample inputs that are e.g. non-contiguous or autograd-leaves, I think we'll lose that with the `clone()` call here?

Maybe the autograd-leaf metadata is unnecessary, but I think primtorch has a `clone_preserve_strides()` util - that might be marginally safer?

Reply:
I agree. In the interest of getting this onto the diff train asap I'm going to merge this first and take a follow-up item.

`clone_preserve_strides` will probably be a bit tricky because it calls as_strided and autograd support on that is weird, I think what we really want is aot_autograd_check to clone the inputs respecting strides and autograd leaves each time the forward is ran.